lib/Conch/Controller/DeviceReport.pm

```diff
@@ -122,7 +122,7 @@ sub process ($c) {
 
     # Time for validations http://www.space.ca/wp-content/uploads/2017/05/giphy-1.gif
     my $validation_plan = $c->_get_validation_plan($unserialized_report);
-    return $c->status(500, { error => 'failed to find validation plan' }) if not $validation_plan;
+    return $c->status(400, { error => 'failed to find validation plan' }) if not $validation_plan;
```

Comment:
I'd probably go with `422` here given how we use it elsewhere. The device report was syntactically valid (otherwise we'd have thrown a validation error at the start, but semantically it contained garbage in the fields for identifying hardware (currently the `device_type`).